File: Data Structures & Algorithms/depth-of-binary-tree/submission-2.py
class Solution(object):
    def maxDepth(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: int
        """
        if not root:
            return 0

        left_depth = self.maxDepth(root.left) + 1
        right_depth = self.maxDepth(root.right) + 1

        return max(left_depth, right_depth)

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-2.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char, int> freq;
        
        int longest = 0;
        int left = 0;
        int max_freq = 0;

        for (int right = 0; right < s.size(); right++) {
            freq[s[right]]++;

            max_freq = max(max_freq, freq[s[right]]);

            while ((right - left + 1) - max_freq > k) {
                freq[s[left]]--;
                left++;
            }

            longest = max(longest, right - left + 1);
        }

        return longest;
    }
};
